conan/internal/api/detect/detect_api.py

```diff
@@ -347,6 +347,54 @@ def detect_cppstd(compiler, compiler_version):
     return cppstd
 
 
+def default_cstd(compiler, compiler_version):
+    """returns the default cstd for the compiler-version. This is not detected, just the default"""
+
+    def _clang_cstd_default(version):
+        if version >= "11":
+            return "gnu17"  # https://releases.llvm.org/11.0.0/tools/clang/docs/ReleaseNotes.html#c-language-changes-in-clang
+        elif version >= "4":  # 3.5 actually
+            return "gnu11"
+        else:
+            return "gnu99"  # It was gnu89 actually
+
+    def _gcc_cstd_default(version):
+        if version >= "15":  # https://www.gnu.org/software/gcc/gcc-15/changes.html#c
+            return "gnu23"
+        elif version >= "8":
+            return "gnu17"  # https://www.gnu.org/software/gcc/gcc-8/changes.html#c
+        elif version >= "5":
+            return "gnu11"  # https://www.gnu.org/software/gcc/gcc-5/changes.html#c
+        else:
+            return "gnu99"  # It was gnu89 actually
+
+    def _visual_cstd_default(version):
+        return None
+
+    def _apple_clang_cstd_default(version):
+        return None
+
+    def _intel_cstd_default(version):
+        return None
+
+    def _mcst_lcc_cstd_default(version):
+        return None
+
+    default = {
+        "gcc": _gcc_cstd_default(compiler_version),
+        "clang": _clang_cstd_default(compiler_version),
+        "apple-clang": _apple_clang_cstd_default(compiler_version),
+        "intel-cc": _intel_cstd_default(compiler_version),
+        "msvc": _visual_cstd_default(compiler_version),
+        "mcst-lcc": _mcst_lcc_cstd_default(compiler_version),
+    }.get(str(compiler), None)
+    return default
+
+
+def detect_cstd(compiler, compiler_version):
```

Comment:
This kind of "alias" wouldn't be necessary, unless Conan ``conan profile detect`` would use a different ``cstd`` than the compiler default in the detection, but as ``conan profile detect`` won't add the ``cstd`` at this moment, this might not be necessary

Reply:
I can remove it, but it might be better to keep it if you decide to add this to conan profile detect in the future. It's up to you.

Reply:
We typically try to follow a "minimalistic" approach, only add what is necessary at the moment, and if in the future the ``detect`` is needed, we can safely add it without breaking.
The other way round can be more problematic, being them equivalent at this moment, some users might mistake the ``detect_cstd`` to retrieve the compiler cstd, not the Conan default-defined, and if we change that behavior in the future, they could consider it breaking, even if it was just their misuse of the function. What is not there cannot be misused.

Reply:
Got it, I will remove